core/src/main/java/org/infinispan/encoding/impl/ConversionManager.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.encoding.impl;
+
+import org.infinispan.commons.dataconversion.ByteArrayWrapper;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.dataconversion.Wrapper;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.Configurations;
+import org.infinispan.configuration.cache.ContentTypeConfiguration;
+import org.infinispan.configuration.cache.EncodingConfiguration;
+import org.infinispan.configuration.cache.StorageType;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.marshall.core.EncoderRegistry;
+import org.infinispan.marshall.persistence.PersistenceMarshaller;
+import org.infinispan.registry.InternalCacheRegistry;
+
+/**
+ * Key/value conversion information.
+ *
+ * @author Dan Berindei
+ * @since 11
+ */
+@Scope(Scopes.NAMED_CACHE)
+public class ConversionManager {
```

Comment:
Not sure about this name, this class is not doing conversion, but providing info about the storage type which does not change for all decorations a cache may have at runtime.

Reply:
I wasn't sure about the name either, WDYT about `StorageManager` or `StorageHelper` @gustavonalle ?

Reply:
```StorageConfigurationManager``` ?

Reply:
Sounds good, updated @gustavonalle